### DISCUSSION TITLE
config help: monorepo patches no longer swept under non-major shared dependencies

### DISCUSSION THREAD

**Discussion by jzempel** [renovatebot/renovate#11732] [Request Help]
In more recent months, it seems that the Renovate's for [Zendesk Garden](https://github.com/zendeskgarden) are churning up a lot more one-off patch PRs than we used to get in the past. I think this can be best illustrated by the bot-generated zendeskgarden/react-components#1189. I took commit control of this PR and added missing patch dependencies in [1233c0a](https://github.com/zendeskgarden/react-components/pull/1189/commits/1233c0aff2ee34b39d2831de9c82f254f90d4502). The resulting merge [autoclosed a lot of the PR noise](https://github.com/zendeskgarden/react-components/pulls?q=is%3Apr+is%3Aclosed+autoclosed) I was hoping to avoid.

Can someone take a peek at our [config](https://github.com/zendeskgarden/react-components/blob/main/.github/renovate.json) and recommend improvements? The associated [Dependency Dashboard](https://github.com/zendeskgarden/react-components/issues/1170) shows config warnings that I'm not exactly sure how to address. Are these related to the problem of excessive patch PRs? The job logs are showing...

```json
WARN: Found renovate config warnings
{
  "warnings": [
    {
      "topic": "Configuration Error",
      "message": "packageRules[2]: Each packageRule must contain at least one non-match* or non-exclude* field. Rule: {\"matchPaths\":[\"packages/**\"]}"
    },
    {
      "topic": "Configuration Error",
      "message": "packageRules[4]: Each packageRule must contain at least one non-match* or non-exclude* field. Rule: {\"matchPaths\":[\"examples/**\"]}"
    }
  ]
}
```

Thanks in advance for any help you can provide here.

**Comment by viceice** [comment]
Looks like the same issue as #11487 which was fixed by #11522

**Comment by viceice** [comment]
this should do it:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "rebaseStalePrs": true,
  "ignoreDeps": ["react-docgen-typescript"],
  "ignorePaths": ["**/node_modules/**"],
  "schedule": ["on Monday every 8 weeks of the year starting on the 5th week"],
  "labels": ["PR: Internal :seedling:"],
  "postUpgradeTasks": {
    "commands": ["yarn install", "yarn format"],
    "fileFilters": ["yarn.lock", "**/*.{js,ts,tsx,md,json}"]
  },
  "packageRules": [
    {
      "matchPaths": ["+(package.json)"],
      "rangeStrategy": "replace"
    },
    {
      "matchPaths": ["+(package.json)"],
      "matchUpdateTypes": ["minor"],
      "groupName": "non-major shared dependencies",
      "groupSlug": "shared-minor-patch"
    },
    {
      "matchPaths": ["packages/**"],
      "matchUpdateTypes": ["minor"],
      "groupName": "non-major package dependencies",
      "groupSlug": "packages-minor-patch"
    },
    {
      "matchPaths": ["examples/**"],
      "matchUpdateTypes": ["minor"],
      "groupName": "non-major example dependencies",
      "groupSlug": "examples-minor-patch"
    },
    {
      "matchPackagePatterns": ["^@zendeskgarden/react"],
      "enabled": false
    },
    {
      "matchDepTypes": ["peerDependencies"],
      "enabled": false
    }
  ]
}
```

**Reply by viceice** [reply]
@rarkins I'm not sure if the first and second rule can be combined, but i think that was the cause for the error?

**Reply by rarkins** [reply]
I've submitted a PR here: https://github.com/zendeskgarden/react-components/pull/1202

**Reply by jzempel** [reply]
wow, so much [cleaner](https://app.renovatebot.com/dashboard#github/zendeskgarden/react-components) now. Thanks a mil @rarkins @viceice.
